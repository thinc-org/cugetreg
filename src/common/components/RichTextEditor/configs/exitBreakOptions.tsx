import { ExitBreakPlugin } from '@udecode/plate-break'
import { KEYS_HEADING } from '@udecode/plate-heading'

export const exitBreakOptions: ExitBreakPlugin = {
  rules: [
    {
      hotkey: 'mod+enter',
    },
    {
      hotkey: 'mod+shift+enter',
      before: true,
    },
    {
      hotkey: 'enter',
      query: {
        start: true,
        end: true,
        allow: KEYS_HEADING,
      },
    },
  ],
}
