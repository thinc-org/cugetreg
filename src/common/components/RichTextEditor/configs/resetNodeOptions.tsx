import { ELEMENT_BLOCKQUOTE } from '@udecode/plate-block-quote'
import { isBlockAboveEmpty, isSelectionAtBlockStart } from '@udecode/plate-core'
import { ELEMENT_PARAGRAPH } from '@udecode/plate-paragraph'
import { ResetNodePlugin } from '@udecode/plate-reset-node'

const resetBlockTypesCommonRule = {
  types: [ELEMENT_BLOCKQUOTE],
  defaultType: ELEMENT_PARAGRAPH,
}

export const resetNodeOptions: ResetNodePlugin = {
  rules: [
    {
      ...resetBlockTypesCommonRule,
      hotkey: 'Enter',
      predicate: isBlockAboveEmpty,
    },
    {
      ...resetBlockTypesCommonRule,
      hotkey: 'Backspace',
      predicate: isSelectionAtBlockStart,
    },
  ],
}
