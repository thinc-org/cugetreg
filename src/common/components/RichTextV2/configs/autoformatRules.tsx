import { autoformatArrow, autoformatLegalHtml, autoformatMath, AutoformatPlugin } from '@udecode/plate'

import { autoformatBlocks } from '../autoformat/block'
import { autoformatLists } from '../autoformat/list'
import { autoformatMarks } from '../autoformat/mark'

export const autoformatOptions: AutoformatPlugin = {
  rules: [
    ...autoformatBlocks,
    ...autoformatLists,
    ...autoformatMarks,
    ...autoformatArrow,
    ...autoformatLegalHtml,
    ...autoformatMath,
  ],
}
