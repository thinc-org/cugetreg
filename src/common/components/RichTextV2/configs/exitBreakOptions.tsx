import { ExitBreakPlugin, KEYS_HEADING } from '@udecode/plate'

export const exitBreakOptions: ExitBreakPlugin = {
  rules: [
    {
      hotkey: 'mod+enter',
    },
    {
      hotkey: 'mod+shift+enter',
      before: true,
    },
    {
      hotkey: 'enter',
      query: {
        start: true,
        end: true,
        allow: KEYS_HEADING,
      },
    },
  ],
}
