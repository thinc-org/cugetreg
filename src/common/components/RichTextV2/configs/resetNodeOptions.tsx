import {
  ELEMENT_BLOCKQUOTE,
  ELEMENT_PARAGRAPH,
  ELEMENT_TODO_LI,
  isBlockAboveEmpty,
  isSelectionAtBlockStart,
  ResetNodePlugin,
} from '@udecode/plate'

const resetBlockTypesCommonRule = {
  types: [ELEMENT_BLOCKQUOTE, ELEMENT_TODO_LI],
  defaultType: ELEMENT_PARAGRAPH,
}

export const resetNodeOptions: ResetNodePlugin = {
  rules: [
    {
      ...resetBlockTypesCommonRule,
      hotkey: 'Enter',
      predicate: isBlockAboveEmpty,
    },
    {
      ...resetBlockTypesCommonRule,
      hotkey: 'Backspace',
      predicate: isSelectionAtBlockStart,
    },
  ],
}
