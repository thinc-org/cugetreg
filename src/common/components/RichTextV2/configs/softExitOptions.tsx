import { ELEMENT_BLOCKQUOTE, ELEMENT_CODE_BLOCK, ELEMENT_TD, SoftBreakPlugin } from '@udecode/plate'

export const softBreakOptions: SoftBreakPlugin = {
  rules: [
    { hotkey: 'shift+enter' },
    {
      hotkey: 'enter',
      query: {
        allow: [ELEMENT_CODE_BLOCK, ELEMENT_BLOCKQUOTE, ELEMENT_TD],
      },
    },
  ],
}
