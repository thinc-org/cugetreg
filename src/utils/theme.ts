import { createMuiTheme, ThemeOptions } from '@material-ui/core'

const lightOptions: ThemeOptions = {
  palette: {
    type: 'light',
    primary: {
      main: '#1890ff',
    },
  },
}

export const lightTheme = createMuiTheme(lightOptions)
