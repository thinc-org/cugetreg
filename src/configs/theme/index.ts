import { deepAssign } from '@/utils/deepAssign'
import { lightPaletteOptions, darkPaletteOptions } from './palette'
import { createMuiTheme, Theme, ThemeOptions } from '@material-ui/core'
import overrideMuiTypography from './typography'
import overrideMuiBaseComponent from './overrides'

const buildTheme = (themeOptions: ThemeOptions): Theme => {
  const theme = createMuiTheme(themeOptions)
  overrideMuiTypography(theme)
  overrideMuiBaseComponent(theme)
  return theme
}

const themeBaseOptions: ThemeOptions = {
  typography: {
    htmlFontSize: 16,
    fontFamily: ['Prompt', 'Poppins', 'sans-serif'].join(','),
  },
  breakpoints: {
    values: {
      xs: 0,
      sm: 600,
      md: 960,
      lg: 1280,
      xl: 1920,
    },
  },
}

const lightThemeOptions: ThemeOptions = deepAssign(themeBaseOptions, {
  palette: lightPaletteOptions,
})

const darkThemeOptions: ThemeOptions = deepAssign(themeBaseOptions, {
  palette: darkPaletteOptions,
})

export const lightTheme = buildTheme(lightThemeOptions)
export const darkTheme = buildTheme(darkThemeOptions)
