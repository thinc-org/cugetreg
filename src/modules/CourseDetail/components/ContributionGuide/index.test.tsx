import { Alert, IconButton } from '@mui/material'
import { shallow } from 'enzyme'

import { lightTheme } from '@/configs/theme'

describe('ContributionGuide', () => {
  const useThemeSpy = jest.fn(() => lightTheme)

  jest.doMock('@mui/material', () => ({
    ...jest.requireActual('@mui/material'),
    useTheme: useThemeSpy,
  }))

  it('should match snapshot', async () => {
    const { ContributionGuide } = await import('.')
    const wrapper = shallow(<ContributionGuide />)
    expect(wrapper).toMatchSnapshot()
  })
})
