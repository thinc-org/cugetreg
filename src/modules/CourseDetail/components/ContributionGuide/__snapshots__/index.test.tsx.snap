// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContributionGuide should match snapshot 1`] = `
<ForwardRef(Alert)
  action={
    <ForwardRef(IconButton)
      onClick={[Function]}
      size="medium"
    >
      <MdKeyboardArrowDown
        color="#0C5A93"
      />
    </ForwardRef(IconButton)>
  }
  icon={false}
  severity="info"
  sx={
    Object {
      "px": 4,
    }
  }
>
  <ForwardRef(Typography)
    color="#0C5A93"
    variant="h6"
  >
    แนวทางในการเขียนรีวิว
  </ForwardRef(Typography)>
  <ForwardRef(Collapse)
    in={false}
  >
    <StyledOL>
      <ForwardRef(Typography)
        color="#0C5A93"
        component="li"
        variant="body1"
      >
        สามารถเล่าประสบการณ์การเรียนในรายวิชาได้ในทุกด้าน เช่น การลงทะเบียน การเรียน การสอบ สิ่งที่อยากจะแนะนำคนอื่น เป็นต้น
      </ForwardRef(Typography)>
      <ForwardRef(Typography)
        color="#0C5A93"
        component="li"
        variant="body1"
      >
        เมื่อรีวิวผ่านการอนุมัติโดยทีมงาน จะปรากฎให้ผู้อื่นเห็นโดยทันที
      </ForwardRef(Typography)>
      <ForwardRef(Typography)
        color="#0C5A93"
        component="li"
        variant="body1"
      >
        ผู้เขียนรีวิวจำเป็นต้องเข้าสู่ระบบด้วยอีเมลนิสิตเพื่อยืนยันความเป็นนิสิตก่อนการรีวิว สามารถดูนโยบายรักษาความเป็นส่วนตัวได้ที่ 
        <Link
          href="/privacy"
        >
          Privacy Policy
        </Link>
         ทั้งนี้ การรีวิวจะเป็นแบบไม่เปิดเผยตัวตนสู่สาธารณะ
      </ForwardRef(Typography)>
      <ForwardRef(Typography)
        color="#0C5A93"
        component="li"
        variant="body1"
      >
        รีวิวที่มีการพูดโจมตีบุคคลอื่น ๆ ด้วยคำพูดหยาบคาย หรือรีวิวที่ทีมงานตัดสินว่ามีลักษณะเป็นสแปม จะไม่ได้รับการอนุมัติ ขอให้แก้ไขเนื้อหาอีกครั้ง
      </ForwardRef(Typography)>
    </StyledOL>
  </ForwardRef(Collapse)>
</ForwardRef(Alert)>
`;
